backend/apps/ifc_validation/views.py

```diff
@@ -34,40 +34,46 @@ class ValidationRequestDetailAPIView(APIView):
     parser_classes = (MultiPartParser, FormParser)
     serializer_class = ValidationRequestSerializer
 
+    def get_for_user_by_public_id(self, user_id, public_id):
+
+        user_requests = ValidationRequest.objects.filter(created_by__id=user_id, deleted=False)
+        instance = [r for r in user_requests if r.public_id == public_id]
```

Comment:
Isn't this wasteful?

Reply:
Since `public_id` is a calculated field, you cannot filter on it.
Can we calculate the `id` from `public_id` somehow?